### PR TITLE
WIP: Add user scope

### PR DESCRIPTION
**Pull request by hugomd**
I'm adding dynamic scopes ✌️ 

TODO:
- [x] Populate User scope
- [x] Use promises
- [x] Clean up
- [x] Properly validate JWT, that user exists and is not deactivated
- [ ] Create a `populators/` folder to store the logic for populating by `userId`, `teamId`, in separate files.

You can [see here](https://blog.andyet.com/2015/06/16/harnessing-hapi-scopes/) for a better idea of what I'm doing.

### CODE REVIEW

**Comment by hugomd** [comment]
Any thoughts @terencehuynh @erfanio ?

**Comment by erfanio** [comment]
LGTM so far
needs more work after we merge team logic

**Comment by hugomd** [comment]
@erfanio I'll need to add authorisation for other resources, and populate for it, but after that it'll be good to merge I think. I'll probably just do users in this PR, merge and handle the conflicts, and then make a new PR for the other populators.

File: app/controllers/Token.js [hunk 1]
```diff
@@ -60,17 +57,18 @@ exports.request = {
...
-          TextBody: `Your token is: ${token.id}`, // TODO: This will need an email template + link to frontend
-        }, () => {
-          // TODO: Logs here
-        });
-
+        console.log(`Token created for ${email}. Token: ${token.id}`);
+        if (env.SEND_EMAIL) { // Only send email in prod
+          // Send the token
+          emailClient.sendEmail({
+            From: 'UNIHACK Team <unihack@anonmail.ovh>',
```
Comment (hugomd) [inline app/controllers/Token.js]: This should be an env variable

File: app/controllers/Token.js [hunk 2]
```diff
@@ -60,17 +57,18 @@ exports.request = {
...
+            To: email,
+            Subject: 'Verify your email address to use UNIHACK',
+            TextBody: `Your token is: ${token.id}`, // TODO: This will need an email template + link to frontend
```
Comment (hugomd) [inline app/controllers/Token.js]: And this should probably be in a template file of some kind :)

**Comment by erfanio** [comment]
@hugomd okay, I can see your point but still I think we shouldn't have `owner` as a scope, rather we should have the `userId`/`teamId`/`inviteId`... as the scope. This way our logic for each route isn't too specific.

And for the privilage level in jwt, I meant that we need some way of overriding scope, so having privilage levels like `normal` or `admin` baked into the jwt is an easy way of doing that. (we give users an admin scope, and all routes allow admin scope)

**Comment by hugomd** [comment]
@erfanio Oh I see what you mean — that sounds like a good idea! 🙌

Theoretically yeah we could do something like that (e.g. [Harnessing Hapi scopes](https://blog.andyet.com/2015/06/16/harnessing-hapi-scopes/) but we'd probably need a permissions collection that maps resources to permissions

**Comment by erfanio** [comment]
@hugomd I'd be good if we could figure out a way to efficiently get all scopes the user has (i.e. `user-1234`, `team-5678`, `invite-1234`, `ticket-5678`) then we wouldn't have to look at the route url to figure out which scopes to inject.

**Pull request closed without merging**